--- ../libbluray.orig/distinfo	2020-04-16 11:59:29.144645000 +0000
+++ ./distinfo	2020-09-30 11:14:15.825302000 +0000
@@ -1,3 +1,3 @@
-TIMESTAMP = 1554632967
-SHA256 (libbluray-1.1.1.tar.bz2) = 106478a17c8bcac8c7b4174e7432b2bad5a1e6dd2926c4224791fd4669472338
-SIZE (libbluray-1.1.1.tar.bz2) = 746034
+TIMESTAMP = 1601464455
+SHA256 (libbluray-1.2.0.tar.bz2) = cd41ea06fd2512a77ebf63872873641908ef81ce2fe4e4c842f6035a47696c11
+SIZE (libbluray-1.2.0.tar.bz2) = 747265
