--- ../dvdauthor.orig/Makefile	2020-10-01 22:09:39.241216000 +0000
+++ ./Makefile	2020-10-01 22:06:59.430698000 +0000
@@ -3,7 +3,7 @@
 
 PORTNAME=	dvdauthor
 PORTVERSION=	0.7.2
-PORTREVISION=	2
+PORTREVISION=	3
 CATEGORIES=	multimedia
 MASTER_SITES=	SF/${PORTNAME}
 
@@ -39,6 +39,7 @@
 PAL_DESC=	Default video format PAL
 
 IMAGEMAGICK_LIB_DEPENDS=	libMagick++-6.so:graphics/ImageMagick6
+IMAGEMAGICK_CFLAGS!=		pkg-config --cflags --libs MagickWand-6
 
 NTSC_CONFIGURE_ON=	--enable-default-video-format=NTSC
 PAL_CONFIGURE_ON=	--enable-default-video-format=PAL
