diff -ruN liba52.orig/Makefile liba52/Makefile
--- liba52.orig/Makefile	2013-10-18 23:49:55.000000000 +0000
+++ liba52/Makefile	2014-02-17 21:04:30.802745127 +0000
@@ -1,5 +1,7 @@
 # Created by: Mario Sergio Fujikawa Ferreira <lioux@FreeBSD.org>
-# $FreeBSD: head/audio/liba52/Makefile 330805 2013-10-18 23:49:55Z bapt $
+# $nuOS$
+# introduced in nuOS 0.0.9.2b2
+# current as of nuOS 0.0.9.2b2
 
 PORTNAME=	liba52
 PORTVERSION=	0.7.4
@@ -46,6 +48,9 @@
 	@${REINPLACE_CMD} -E -e 's|(fftc4.h)|djbfft/\1|' ${WRKSRC}/liba52/imdct.c
 .endif
 
+pre-install:
+	${FIND} ${WRKSRC} -name Makefile | ${XARGS} ${TOUCH}
+
 post-install:
 	${MKDIR} ${STAGEDIR}${DOCSDIR}
 	${INSTALL_DATA} ${WRKSRC}/doc/liba52.txt ${STAGEDIR}${DOCSDIR}
