--- ../talloc.orig/distinfo	2019-11-19 16:53:05.910129000 +0000
+++ ./distinfo	2019-11-17 02:45:27.060929000 +0000
@@ -1,3 +1,3 @@
-TIMESTAMP = 1558143287
-SHA256 (talloc-2.2.0.tar.gz) = 5c6f6a45ef96b3fd0b28942673a68d0c6af5dcca9d676a2e4d57ce7e86c22ebc
-SIZE (talloc-2.2.0.tar.gz) = 633467
+TIMESTAMP = 1573498827
+SHA256 (talloc-2.3.0.tar.gz) = 75d5bcb34482545a82ffb06da8f6c797f963a0da450d0830c669267b14992fc6
+SIZE (talloc-2.3.0.tar.gz) = 642364
