--- ../knot2.orig/Makefile 2018-01-18 15:05:57.000000000 +0000
+++ ./Makefile      2018-01-20 21:26:57.856745000 +0000
@@ -2,7 +2,7 @@
 # $FreeBSD: head/dns/knot2/Makefile 453673 2017-11-07 19:15:47Z matthew $
 
 PORTNAME=      knot
-PORTVERSION=   2.6.1
+PORTVERSION=   2.6.4
 CATEGORIES=    dns ipv6
 MASTER_SITES=  https://secure.nic.cz/files/knot-dns/ \
                http://dns-lab.com/downloads/knot-dns/
