--- ../blender.orig/files/patch-build_files_cmake_Modules_FindPythonLibsUnix.cmake	2020-09-10 12:42:09.692553000 +0000
+++ ./files/patch-build_files_cmake_Modules_FindPythonLibsUnix.cmake	2021-04-10 16:36:32.364645000 +0000
@@ -5,7 +5,7 @@
  ENDIF()
 
 -SET(PYTHON_VERSION 3.7 CACHE STRING "Python Version (major and minor only)")
-+SET(PYTHON_VERSION 3.8 CACHE STRING "Python Version (major and minor only)")
++SET(PYTHON_VERSION 3.9 CACHE STRING "Python Version (major and minor only)")
  MARK_AS_ADVANCED(PYTHON_VERSION)
 
 
