--- ../tiff.orig/files/patch-tif_getimage.c	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-tif_getimage.c	2017-05-20 15:57:17.881933000 +0000
@@ -0,0 +1,11 @@
+--- libtiff/tif_getimage.c.orig	2016-11-18 02:47:45 UTC
++++ libtiff/tif_getimage.c
+@@ -1305,7 +1305,7 @@ DECLAREContigPutFunc(putagreytile)
+     while (h-- > 0) {
+ 	for (x = w; x-- > 0;)
+         {
+-            *cp++ = BWmap[*pp][0] & (*(pp+1) << 24 | ~A1);
++            *cp++ = BWmap[*pp][0] & ((uint32)*(pp+1) << 24 | ~A1);
+             pp += samplesperpixel;
+         }
+ 	cp += toskew;
