--- ../tiff.orig/files/patch-tif_jpeg.c	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-tif_jpeg.c	2017-05-20 15:57:17.884062000 +0000
@@ -0,0 +1,16 @@
+--- libtiff/tif_jpeg.c.orig	2017-05-20 15:31:27 UTC
++++ libtiff/tif_jpeg.c
+@@ -1632,6 +1632,13 @@ JPEGSetupEncode(TIFF* tif)
+                             "Invalig horizontal/vertical sampling value");
+                     return (0);
+                 }
++                if( td->td_bitspersample > 16 )
++                {
++                    TIFFErrorExt(tif->tif_clientdata, module,
++                                 "BitsPerSample %d not allowed for JPEG",
++                                 td->td_bitspersample);
++                    return (0);
++                }
+ 
+ 		/*
+ 		 * A ReferenceBlackWhite field *must* be present since the
