--- ../tiff.orig/files/patch-tif_unix.c	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-tif_unix.c	2017-05-20 15:57:17.890438000 +0000
@@ -0,0 +1,17 @@
+--- libtiff/tif_unix.c.orig	2015-08-28 22:16:22 UTC
++++ libtiff/tif_unix.c
+@@ -316,6 +316,14 @@ _TIFFmalloc(tmsize_t s)
+ 	return (malloc((size_t) s));
+ }
+ 
++void* _TIFFcalloc(tmsize_t nmemb, tmsize_t siz)
++{
++    if( nmemb == 0 || siz == 0 )
++        return ((void *) NULL);
++
++    return calloc((size_t) nmemb, (size_t)siz);
++}
++
+ void
+ _TIFFfree(void* p)
+ {
