--- ../tiff.orig/files/patch-tools_tiffcrop.c	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-tools_tiffcrop.c	2017-05-20 15:57:17.900888000 +0000
@@ -0,0 +1,10 @@
+--- tools/tiffcrop.c.orig	2016-11-19 01:45:30 UTC
++++ tools/tiffcrop.c
+@@ -7986,7 +7986,6 @@ writeCroppedImage(TIFF *in, TIFF *out, s
+   if (!TIFFWriteDirectory(out))
+     {
+     TIFFError("","Failed to write IFD for page number %d", pagenum);
+-    TIFFClose(out);
+     return (-1);
+     }
+ 
