--- ../tiff.orig/files/patch-tiffio.h	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-tiffio.h	2017-05-20 15:57:17.894654000 +0000
@@ -0,0 +1,10 @@
+--- libtiff/tiffio.h.orig	2016-01-24 15:39:51 UTC
++++ libtiff/tiffio.h
+@@ -293,6 +293,7 @@ extern TIFFCodec* TIFFGetConfiguredCODEC
+  */
+ 
+ extern void* _TIFFmalloc(tmsize_t s);
++extern void* _TIFFcalloc(tmsize_t nmemb, tmsize_t siz);
+ extern void* _TIFFrealloc(void* p, tmsize_t s);
+ extern void _TIFFmemset(void* p, int v, tmsize_t c);
+ extern void _TIFFmemcpy(void* d, const void* s, tmsize_t c);
