--- ../icedtea-web.orig/pkg-plist	2018-06-25 03:47:39.377774000 +0000
+++ pkg-plist	2018-06-23 12:29:23.013316000 +0000
@@ -1,6 +1,7 @@
 bin/itweb-javaws
 bin/itweb-policyeditor
 bin/itweb-settings
+%%PLUGIN%%%%WEBPLUGIN_DIR%%/IcedTeaPlugin.so
 man/cs/man1/icedtea-web-plugin.1.gz
 man/cs/man1/icedtea-web.1.gz
 man/cs/man1/itweb-javaws.1.gz
@@ -26,4 +27,5 @@
 share/applications/itweb-settings.desktop
 %%DATADIR%%/javaws_splash.png
 %%DATADIR%%/netx.jar
+%%PLUGIN%%%%DATADIR%%/plugin.jar
 share/pixmaps/itweb-javaws.png
