--- /dev/null	2018-06-26 20:07:35.000000000 -0700
+++ files/patch-Modules__uuidmodule.c	2018-07-06 00:27:09.979104000 -0700
@@ -0,0 +1,12 @@ 
+--- Modules/_uuidmodule.c.orig	2018-06-26 20:07:35.000000000 -0700
++++ Modules/_uuidmodule.c	2018-07-06 00:27:09.979104000 -0700
+@@ -1,9 +1,6 @@
+ #define PY_SSIZE_T_CLEAN
+ 
+ #include "Python.h"
+-#ifdef HAVE_UUID_UUID_H
+-#include <uuid/uuid.h>
+-#endif
+ #ifdef HAVE_UUID_H
+ #include <uuid.h>
+ #endif
