diff -ruN lua53.orig/Makefile lua53/Makefile
--- lua53.orig/Makefile	2015-06-27 19:37:35.970893000 +0000
+++ lua53/Makefile	2015-07-10 03:59:31.803891784 +0000
@@ -2,7 +2,7 @@
 # $FreeBSD: head/lang/lua53/Makefile 390568 2015-06-25 13:12:20Z emaste $
 
 PORTNAME=	lua
-PORTVERSION=	5.3.0
+PORTVERSION=	5.3.1
 CATEGORIES=	lang
 MASTER_SITES=	http://www.lua.org/ftp/
 PKGNAMESUFFIX=	53
diff -ruN lua53.orig/distinfo lua53/distinfo
--- lua53.orig/distinfo	2015-01-17 02:19:59.000000000 +0000
+++ lua53/distinfo	2015-07-10 03:59:31.803891784 +0000
@@ -1,2 +1,2 @@
-SHA256 (lua-5.3.0.tar.gz) = ae4a5eb2d660515eb191bfe3e061f2b8ffe94dce73d32cfd0de090ddcc0ddb01
-SIZE (lua-5.3.0.tar.gz) = 278045
+SHA256 (lua-5.3.1.tar.gz) = 072767aad6cc2e62044a66e8562f51770d941e972dc1e4068ba719cd8bffac17
+SIZE (lua-5.3.1.tar.gz) = 282401
