--- ../postfix.orig/distinfo	2018-02-13 13:20:37.441605000 +0000
+++ ./distinfo	2018-02-07 18:00:45.991629000 +0000
@@ -1,3 +1,3 @@
-TIMESTAMP = 1517758440
-SHA256 (postfix/postfix-3.3.0-RC1.tar.gz) = 8bf16bee213bb5de95cbcab30479f3fd1c7fd2940fd9b4e63a1a6104c1fe3658
-SIZE (postfix/postfix-3.3.0-RC1.tar.gz) = 4418769
+TIMESTAMP = 1517107020
+SHA256 (postfix/postfix-3.2.5.tar.gz) = 7d0b841d3a46e7c64f392fcc4de5dd3f4bdcfd0b9e48fc235bdff6a208f0f376
+SIZE (postfix/postfix-3.2.5.tar.gz) = 4394765
