--- ../namecoin.orig/Makefile	2018-09-28 12:34:17.304721000 +0000
+++ ./Makefile	2018-10-28 12:41:27.978731000 +0000
@@ -2,7 +2,7 @@
 # $FreeBSD: head/net-p2p/namecoin/Makefile 479111 2018-09-06 18:06:51Z rene $
 
 PORTNAME=	namecoin
-PORTVERSION=	0.16.2
+PORTVERSION=	0.17.0
 DISTVERSIONPREFIX=	nc
 PORTEPOCH=	1
 CATEGORIES=	net-p2p dns
@@ -102,10 +102,8 @@
 .if defined(SLAVE_PORT) && ${SLAVE_PORT} == "no"
 
 post-install:
-	${REINPLACE_CMD} -e 's/bitcoin/namecoin/g;s/Bitcoin/Namecoin/g' \
-		${WRKSRC}/contrib/debian/bitcoin-qt.desktop
-	${INSTALL} ${WRKSRC}/contrib/debian/bitcoin-qt.desktop \
-		${STAGEDIR}${PREFIX}/share/applications/namecoin-qt.desktop
+	${INSTALL} ${FILESDIR}/namecoin-qt.desktop \
+		${STAGEDIR}${PREFIX}/share/applications/
 	${LOCALBASE}/bin/svg2png -w128 -h128 ${WRKSRC_graphics}/svg/namecoin-logo.svg \
 		${STAGEDIR}${PREFIX}/share/pixmaps/namecoin128.png
 
