--- ../namecoin.orig/Makefile	2020-06-05 04:05:52.618568000 +0000
+++ ./Makefile	2020-06-05 03:15:28.971036000 +0000
@@ -2,8 +2,7 @@
 # $FreeBSD: head/net-p2p/namecoin/Makefile 537830 2020-06-03 18:32:05Z sunpoet $
 
 PORTNAME=	namecoin
-PORTVERSION=	0.19.0.1
-PORTREVISION=	2
+PORTVERSION=	0.19.1
 DISTVERSIONPREFIX=	nc
 PORTEPOCH=	1
 CATEGORIES=	net-p2p dns
@@ -11,17 +10,11 @@
 MAINTAINER=	milios@ccsys.com
 COMMENT?=	Decentralized, open DNS and general purpose key/value store
 
-DEPRECATED=	Relies on implementation details of a particular Berkeley DB version (4.8), which has been EOL since 2010.
-EXPIRATION_DATE=2020-06-30
-
 LICENSE=	MIT
 
 LIB_DEPENDS=	libboost_date_time.so:devel/boost-libs \
 		libevent.so:devel/libevent
 
-CONFLICTS_INSTALL=	namecoin-beta-[0-9]* namecoin-beta-daemon-[0-9]* namecoin-beta-utils-[0-9]* \
-		namecoin-stable-[0-9]* namecoin-stable-daemon-[0-9]* namecoin-stable-utils-[0-9]*
-
 USES=		autoreconf compiler:c++11-lib gmake libtool pkgconfig ssl
 GNU_CONFIGURE=	yes
 
@@ -79,10 +72,14 @@
 QRCODES_CONFIGURE_WITH=	qrencode
 QRCODES_LIB_DEPENDS=	libqrencode.so:graphics/libqrencode
 
-WALLET_CONFIGURE_ENABLE=	wallet
-WALLET_CXXFLAGS=		-I${BDB_INCLUDE_DIR}
-WALLET_LIBS=			-L${BDB_LIB_DIR}
-WALLET_USES=			bdb
+WALLET_CONFIGURE_OFF=	--disable-wallet
+WALLET_CONFIGURE_ON=	--enable-wallet \
+			--with-incompatible-bdb \
+			BDB_CFLAGS=-I${BDB_INCLUDE_DIR} \
+			BDB_LIBS="-L${BDB_LIB_DIR} \
+				-Wl,-rpath=${BDB_LIB_DIR} \
+				-l${BDB_LIB_CXX_NAME}"
+WALLET_USES=		bdb:5+
 
 ZMQ_CONFIGURE_ENABLE=	zmq
 ZMQ_BUILD_DEPENDS=	libzmq4>0:net/libzmq4
@@ -118,10 +115,4 @@
 
 	@cd ${WRKSRC} && ${GMAKE} check
 
-.include <bsd.port.options.mk>
-
-.if !empty(PORT_OPTIONS:MWALLET)
-BROKEN=	Critical design flaw: Wallet option relies on Berkeley DB 4.8 implementation internals, where Berkeley DB 4.8 has been EOL since 2010
-.endif
-
 .include <bsd.port.mk>
