--- ../namecoin.orig/distinfo	2020-06-05 04:05:52.620774000 +0000
+++ ./distinfo	2020-06-05 03:15:54.535459000 +0000
@@ -1,5 +1,5 @@
-TIMESTAMP = 1581654464
-SHA256 (namecoin-namecoin-core-nc0.19.0.1_GH0.tar.gz) = a9d403822a7ff470f9bf1467e6833af53af7c80a8c696eb259f16b105cba7ad3
-SIZE (namecoin-namecoin-core-nc0.19.0.1_GH0.tar.gz) = 5873509
+TIMESTAMP = 1591326954
+SHA256 (namecoin-namecoin-core-nc0.19.1_GH0.tar.gz) = afd788e6ffbea3ab01c8657a51f3fcc061478618225adebb8dd8c0aecf780023
+SIZE (namecoin-namecoin-core-nc0.19.1_GH0.tar.gz) = 5904070
 SHA256 (indolering-namecoin-graphics-a13a73b_GH0.tar.gz) = 82f308a3527d96a202099feb96a754d036bbdaa875156ad002ba2747d75000c7
 SIZE (indolering-namecoin-graphics-a13a73b_GH0.tar.gz) = 7364908
