diff -ru namecoin.orig/Makefile namecoin/Makefile
--- namecoin.orig/Makefile	2015-05-07 17:26:05.000000000 +0000
+++ namecoin/Makefile	2015-05-15 02:27:07.318885104 +0000
@@ -2,14 +2,16 @@
 # $FreeBSD: head/net-p2p/namecoin/Makefile 385640 2015-05-07 17:26:05Z mat $
 
 PORTNAME=	namecoin
-PORTVERSION=	Q.3.72
-DISTVERSIONPREFIX=	v
-PORTREVISION=	3
+PORTVERSION=	0.3.75
+PORTEPOCH=	1
+DISTVERSIONPREFIX=	nc
 CATEGORIES=	net-p2p dns
 
 MAINTAINER=	milios@ccsys.com
 COMMENT=	Decentralized, open DNS system and general purpose key/value store
 
+LICENSE=	MIT
+
 LIB_DEPENDS=	libboost_date_time.so:${PORTSDIR}/devel/boost-libs
 
 OPTIONS_DEFINE=	X11 UPNP QRCODES DBUS
@@ -19,8 +21,8 @@
 QRCODES_DESC=	Build with QR code display
 
 USE_GITHUB=	yes
-GH_ACCOUNT=	namecoinq
-GH_PROJECT=	namecoinq
+GH_ACCOUNT=	namecoin
+GH_PROJECT=	namecoin
 
 USES=		gmake compiler:c++11-lib
 USE_OPENSSL=	yes
@@ -62,6 +64,7 @@
 
 .if ${PORT_OPTIONS:MUPNP}
 LIB_DEPENDS+=	libminiupnpc.so:${PORTSDIR}/net/miniupnpc
+CXXFLAGS+=	-I${LOCALBASE}/include/miniupnpc
 QMAKE_USE_UPNP=	1
 .else
 QMAKE_USE_UPNP=	-
diff -ru namecoin.orig/distinfo namecoin/distinfo
--- namecoin.orig/distinfo	2015-05-07 17:26:05.000000000 +0000
+++ namecoin/distinfo	2015-05-15 02:27:55.917892500 +0000
@@ -1,2 +1,2 @@
-SHA256 (namecoinq-namecoinq-vQ.3.72_GH0.tar.gz) = 74d088f96735b3397b9a71bc10bbbe1211cd2594baf593c4b9062457c1d4b869
-SIZE (namecoinq-namecoinq-vQ.3.72_GH0.tar.gz) = 1453901
+SHA256 (namecoin-namecoin-nc0.3.75_GH0.tar.gz) = 6ca938951f408828a3b3d2c3ec3078956aa8a060405eb8f0632f4bfd824d7f98
+SIZE (namecoin-namecoin-nc0.3.75_GH0.tar.gz) = 1464385
