--- /dev/null	2018-06-26 13:55:00.000000000 +0000
+++ files/patch-ompi_contrib_vt_vt_vtlib_vt__mallocwrap.c	2018-06-26 13:48:30.007794000 +0000
@@ -0,0 +1,13 @@
+--- ompi/contrib/vt/vt/vtlib/vt_mallocwrap.c.orig	2018-06-26 13:47:41 UTC
++++ ompi/contrib/vt/vt/vtlib/vt_mallocwrap.c
+@@ -26,10 +26,6 @@
+ 
+ #include <errno.h>
+ #include <stdlib.h>
+-#if (defined(HAVE_MEMALIGN) && HAVE_MEMALIGN) || \
+-    (defined(HAVE_VALLOC) &&  HAVE_VALLOC)
+-# include <malloc.h>
+-#endif /* HAVE_MEMALIGN || HAVE_VALLOC */
+ 
+ 
+ /* define the following macro to enable tracing the calloc function which
