--- ../ipxe.orig/Makefile	2018-06-09 21:29:27.297241000 +0000
+++ Makefile	2018-06-10 10:48:40.421276000 +0000
@@ -56,12 +56,15 @@
 _IPXE_BUILDCFG=	branding:PRODUCT_NAME="${PKGNAME} (${OPSYS} ${OSREL}/${ARCH})" \
 		console:CONSOLE_SERIAL \
 		console:KEYBOARD_MAP=us \
-		general:!DOWNLOAD_PROTO_FTP \
+		general:DOWNLOAD_PROTO_FTP \
 		general:DOWNLOAD_PROTO_HTTPS \
 		general:DOWNLOAD_PROTO_NFS \
 		general:IMAGE_TRUST_CMD \
+		general:CERT_CMD \
+		general:DIGEST_CMD \
 		general:NET_PROTO_IPV6 \
 		general:PING_CMD \
+		general:PCI_CMD \
 		general:POWEROFF_CMD
 
 OPTIONS_DEFAULT=	ISO EFI
