diff -ruN ../libvncserver.orig/files/patch-libvncclient_sasl.c ./files/patch-libvncclient_sasl.c
--- ../libvncserver.orig/files/patch-libvncclient_sasl.c	1970-01-01 00:00:00.000000000 +0000
+++ ./files/patch-libvncclient_sasl.c	2019-01-30 03:43:48.500282000 +0000
@@ -0,0 +1,11 @@
+--- libvncclient/sasl.c.orig	2019-01-30 03:43:33 UTC
++++ libvncclient/sasl.c
+@@ -55,6 +55,8 @@
+ #endif /* LIBVNCSERVER_HAVE_WS2TCPIP_H */
+ #else /* WIN32 */
+ #include <arpa/inet.h>
++#include <netinet/in.h>
++#include <sys/socket.h>
+ #endif /* WIN32 */
+ 
+ #include "sasl.h"
