diff -ruN ../tor.orig/Makefile ./Makefile
--- ../tor.orig/Makefile	2015-10-17 22:00:42.709115000 +0000
+++ ./Makefile	2015-12-12 02:35:45.655116498 +0000
@@ -2,7 +2,7 @@
 # $FreeBSD: head/security/tor/Makefile 399346 2015-10-15 14:55:14Z mat $
 
 PORTNAME=	tor
-DISTVERSION=	0.2.6.10
+DISTVERSION=	0.2.7.6
 CATEGORIES=	security net ipv6
 MASTER_SITES=	TOR
 
@@ -44,6 +44,10 @@
 
 .include <bsd.port.options.mk>
 
+.if ${OSVERSION} < 1000000
+WITH_OPENSSL_PORT=	yes
+.endif
+
 .if !defined(USE_GCC) && empty(CC:T:M*gcc4*) && \
 empty(PORT_OPTIONS:MSTATIC_TOR) && empty(ARCH:Mia64)
 CONFIGURE_ARGS+=	--enable-gcc-hardening
diff -ruN ../tor.orig/distinfo ./distinfo
--- ../tor.orig/distinfo	2015-07-13 21:24:24.000000000 +0000
+++ ./distinfo	2015-12-12 02:35:45.657118243 +0000
@@ -1,2 +1,2 @@
-SHA256 (tor-0.2.6.10.tar.gz) = 0542c0efe43b86619337862fa7eb02c7a74cb23a79d587090628a5f0f1224b8d
-SIZE (tor-0.2.6.10.tar.gz) = 3587211
+SHA256 (tor-0.2.7.6.tar.gz) = 493a8679f904503048114aca6467faef56861206bab8283d858f37141d95105d
+SIZE (tor-0.2.7.6.tar.gz) = 4869754
