diff -ruN tor.orig/Makefile tor/Makefile
--- tor.orig/Makefile	2015-06-14 02:10:57.270884000 +0000
+++ tor/Makefile	2015-07-13 21:22:04.171361379 +0000
@@ -2,7 +2,7 @@
 # $FreeBSD: head/security/tor/Makefile 389268 2015-06-12 13:11:38Z bf $
 
 PORTNAME=	tor
-DISTVERSION=	0.2.6.9
+DISTVERSION=	0.2.6.10
 CATEGORIES=	security net ipv6
 MASTER_SITES=	TOR
 
diff -ruN tor.orig/distinfo tor/distinfo
--- tor.orig/distinfo	2015-06-14 02:10:56.857881000 +0000
+++ tor/distinfo	2015-07-13 21:24:24.011365497 +0000
@@ -1,2 +1,2 @@
-SHA256 (tor-0.2.6.9.tar.gz) = 4a6c29ad89a98d7832c599d9480d6d8e55355fb3b8f4b506c5df557f15942f9c
-SIZE (tor-0.2.6.9.tar.gz) = 3581981
+SHA256 (tor-0.2.6.10.tar.gz) = 0542c0efe43b86619337862fa7eb02c7a74cb23a79d587090628a5f0f1224b8d
+SIZE (tor-0.2.6.10.tar.gz) = 3587211
