--- ../youtube_dl.orig/Makefile	2018-11-12 11:34:53.744685000 +0000
+++ Makefile	2018-11-12 11:29:15.190621000 +0000
@@ -1,8 +1,7 @@
 # $FreeBSD: head/www/youtube_dl/Makefile 484273 2018-11-06 01:50:26Z jbeich $
 
 PORTNAME=	youtube_dl
-PORTVERSION=	2018.09.10
-PORTREVISION=	1
+PORTVERSION=	2018.11.07
 CATEGORIES=	www
 MASTER_SITES=	https://yt-dl.org/downloads/${PORTVERSION}/
 DISTNAME=	youtube-dl-${PORTVERSION}
