--- ../youtube_dl.orig/Makefile	2020-09-28 22:12:38.123145000 +0000
+++ ./Makefile	2020-11-02 17:43:34.013777000 +0000
@@ -1,9 +1,9 @@
 # $FreeBSD: head/www/youtube_dl/Makefile 550400 2020-09-28 05:05:24Z tobik $
 
 PORTNAME=	youtube_dl
-PORTVERSION=	2020.09.20
+PORTVERSION=	2020.11.01.1
 CATEGORIES=	www
-MASTER_SITES=	https://yt-dl.org/downloads/${PORTVERSION}/
+MASTER_SITES=	https://youtube-dl.org/downloads/latest/
 DISTNAME=	youtube-dl-${PORTVERSION}
 
 # Implicit approval to commit trivial version updates.
