diff -ru nginx.bak/Makefile nginx/Makefile
--- nginx.bak/Makefile	2014-12-12 22:51:48.000000000 +0000
+++ nginx/Makefile	2014-12-23 18:06:07.916310467 +0000
@@ -701,11 +701,10 @@
 .endif
 
 .if ${PORT_OPTIONS:MRTMP}
-NGINX_RTMP_VERSION=	1.1.5
+NGINX_RTMP_VERSION=	1.1.6
 MASTER_SITES+=	https://github.com/arut/nginx-rtmp-module/archive/v${NGINX_RTMP_VERSION}/:rtmp
 DISTFILES+=	rtmp-nginx-module-v${NGINX_RTMP_VERSION}.tar.gz:rtmp
 CONFIGURE_ARGS+=--add-module=${WRKDIR}/nginx-rtmp-module-${NGINX_RTMP_VERSION}
-EXTRA_PATCHES+=	${PATCHDIR}/extra-patch-arut-rtmp-module
 .endif
 
 .if ${PORT_OPTIONS:MSET_MISC}
diff -ru nginx.bak/distinfo nginx/distinfo
--- nginx.bak/distinfo	2014-10-04 14:16:16.000000000 +0000
+++ nginx/distinfo	2014-12-23 16:43:47.834306634 +0000
@@ -86,8 +86,8 @@
 SIZE (openresty-redis2-nginx-module-v0.10-0-g78a7622.tar.gz) = 29850
 SHA256 (alibaba-nginx-footer-filter-1.2.2-0-g3572184.tar.gz) = 48a23f879416c0d8da5795c419b94f1ffa42ee06602545f494db4d9aba646e8f
 SIZE (alibaba-nginx-footer-filter-1.2.2-0-g3572184.tar.gz) = 3934
-SHA256 (rtmp-nginx-module-v1.1.5.tar.gz) = 990467fed853b8eecd9b1b10050e2406071d705896be452539be8eced24adf4a
-SIZE (rtmp-nginx-module-v1.1.5.tar.gz) = 519507
+SHA256 (rtmp-nginx-module-v1.1.6.tar.gz) = 4039d1e7febd93188f729b594772d04d8a1137b2e90b12fa53bb061f200add87
+SIZE (rtmp-nginx-module-v1.1.6.tar.gz) = 519666
 SHA256 (openresty-set-misc-nginx-module-v0.24-0-g36fd035.tar.gz) = 5475877d67a952beaf0b726b96d0c9abf48ccfc27367a6bebb5738e04d054a76
 SIZE (openresty-set-misc-nginx-module-v0.24-0-g36fd035.tar.gz) = 40390
 SHA256 (nginx-sflow-module-0.9.7.tar.gz) = 508d15a43059abab08281bfa66d2dd520e2e7635d0b17043549bb331c8755b76
Only in nginx.bak/files: extra-patch-arut-rtmp-module
